### PR TITLE
Table view changes for this week

### PR DESCRIPTION
**Pull request by taylerhow**
This PR addresses several JIRA cards:
- [LED-136] Add Transaction from within table view
- [LED-144] Optional Debit/Credit perspective
- [LED-151] Remove currency icon ($)
- [LED-152] Option Check Number column

Additional changes:
- Updated the Amount string converters to work correctly
- Updated the table view to support horizontal scrolling

Please test this branch very thoroughly. Consider the following basic operations (and edge cases derived from them):
- Adding a transaction from within the table view, and all of the related behavior
- Switching between regular Amount and Debit/Credit perspective
- Sorting by Debit/Credit
- Editing Debit/Credit columns
- Editing check numbers
- Sorting by check numbers
- Look for any instances of $ within the app
- Test what kind of input is valid when editing amounts
- Toggling display for optional columns like Check Number, Account, and Tags

### CODE REVIEW

File: src/main/java/ledger/user_interface/ui_controllers/component/UserTransactionInput.java [hunk 1]
```diff
@@ -161,7 +161,7 @@ public Transaction getTransactionSubmission() {
         }
 
         Transaction t = new Transaction(date, type, amount, account,
-                payee, pending, category, notes);
+                payee, pending, category, notes, Integer.parseInt(checkNo));
```
Comment (xxshadowfang) [inline src/main/java/ledger/user_interface/ui_controllers/component/UserTransactionInput.java]: So This works when there is a check but it doesnt if a different type is selected. You get a invalid number exception. Probably an else statement above setting it to -1 or something would work.

File: src/main/java/ledger/user_interface/ui_controllers/component/TransactionTableView.java [hunk 2]
```diff
@@ -34,8 +38,20 @@
 public class TransactionTableView extends TableView<Transaction> implements IUIController, Initializable {
```
Comment (xxshadowfang) [inline src/main/java/ledger/user_interface/ui_controllers/component/TransactionTableView.java]: any reason there are two Category/tags columns that do the same thing?

File: src/main/java/ledger/user_interface/ui_controllers/component/UserTransactionInput.java (hunk 1 above)
Comment (shellajt) [inline src/main/java/ledger/user_interface/ui_controllers/component/UserTransactionInput.java]: This is fixed in my branch

Comment (shellajt) [inline src/main/java/ledger/user_interface/ui_controllers/component/UserTransactionInput.java]: My branch is merged in, this is fixed

File: src/main/java/ledger/user_interface/ui_controllers/component/TransactionTableView.java (hunk 2 above)
Comment (taylerhow) [inline src/main/java/ledger/user_interface/ui_controllers/component/TransactionTableView.java]: I fixed this in commit 00175bc, see below.

**Comment by taylerhow** [comment]
@LadyMopar I addressed the null pointer issue in commit 9cc293f, see below you change request. As for scrolling, horizontal scrolling is disabled right now. I'm testing some options locally but none of them are really working out.

**Comment by LadyMopar** [comment]
I can't seem to break it. I did notice one thing that doesn't really fit into this PR I think - if you have an empty table and you right-click to delete a transaction, a popup shows up to ask if you want to delete the selected transactions, and when you click ok, it acts like nothing's wrong. I think a better popup for this edge case would be something along the lines of "no transaction selected" or something.
